addOns/automation/src/main/java/org/zaproxy/addon/automation/jobs/RequestorJob.java

```diff
@@ -147,13 +158,28 @@ public void runJob(AutomationEnvironment env, AutomationProgress progress) {
                 msg.getRequestBody().setBody(env.replaceVars(req.getData()));
                 msg.getRequestHeader().setContentLength(msg.getRequestBody().length());
             }
+            User user = this.getUser(this.getParameters().getUser(), progress);
+
             try {
-                progress.info(
-                        Constant.messages.getString(
-                                "automation.info.requrl",
-                                this.getName(),
-                                msg.getRequestHeader().getURI()));
-                httpSender.sendAndReceive(msg);
+                if (user != null) {
+                    httpSender.setUser(user);
```

Comment:
The user could be set in the message, would avoid the need to set/reset each time (same for spider job).

Reply:
Done :)